Signup and Login routes done

Kysumi opened the pull request:
Need to add validation to the signup for duplicate emails ect.

Not sure on how we want to implement this, so have left it for now.

Kysumi committed:
Signup and Login routes done

Need to add validation to the signup for duplicate emails ect.

Kysumi committed:
Clean up and adding types

Kysumi committed:
auth working correctly

jallier commented on backend/src/auth/passport.ts:
```diff
@@ -0,0 +1,90 @@
+    if (!user) {
+      // If the user isn't found in the database, return a message
+      return done(null, false, {
+        message: 'User not found'
```
We should probably change this later so that it doesn't tell anyone if the account already exists. It should just say 'Either user or password is incorrect'.

But we can change that later

Kysumi replied:
Yip this was mainly for testing. I haven't spent time on the validation yet as I'm not how we want to go about it.

jallier commented on backend/src/auth/passport.ts:
```diff
@@ -0,0 +1,90 @@
+// This verifies that the token sent by the user is valid
+passport.use(new JWTstrategy({
+  // move to dotenv!
+  secretOrKey: 'ezybuccccccs',
```
I know we need to make this an env variable, but it might be worth making it a variable in the mean time, that way we only need to change one line later on once we change that.

Kysumi replied:
Yea we can do this. I commented dotenv above so we can find these later to swap them

jallier commented on backend/src/route/auth.route.ts:
```diff
@@ -1,30 +1,64 @@
+                    // sign JWT! need to move this into dotenv!
+                    const token = jwt.sign({
+                        user: body
+                    }, 'ezybuccccccs');
```
Like here, we can refer to the variable instead of a hard coded string

jallier reviewed (changes requested):
I think we should just change the jwt secret to be a variable to make it easier to change in the future

Kysumi committed:
Added pre install hook

set PM2 to use 2 cores
moved "secret" into config file to change at later date.

changed login validation error.
